crates/astria-gossipnet/src/network_stream.rs

```diff
@@ -0,0 +1,271 @@
+                }
+                SwarmEvent::ConnectionEstablished {
+                    peer_id,
+                    endpoint: _,
+                    num_established,
+                    concurrent_dial_errors: _,
+                    established_in: _,
+                } => {
+                    debug!(
+                        "Connection with {peer_id} established (total: {num_established})",
+                        peer_id = peer_id,
+                        num_established = num_established,
+                    );
+                    self.swarm
+                        .behaviour_mut()
+                        .gossipsub
+                        .add_explicit_peer(&peer_id);
+
+                    return Poll::Ready(Some(Event::PeerConnected(peer_id)));
+                }
+
+                SwarmEvent::Behaviour(GossipnetBehaviourEvent::Identify(
+                    libp2p::identify::Event::Received {
+                        peer_id,
+                        info,
+                    },
+                )) => {
+                    debug!(
+                        "Received identify event from {peer_id:?} with info: {info:?}",
+                        peer_id = peer_id,
+                        info = info,
+                    );
+                    for addr in info.listen_addrs {
+                        self.swarm
+                            .behaviour_mut()
+                            .kademlia
+                            .add_address(&peer_id, addr);
+                    }
+                }
+
+                // DHT events
+                #[cfg(feature = "dht")]
+                SwarmEvent::Behaviour(GossipnetBehaviourEvent::Kademlia(
+                    KademliaEvent::RoutingUpdated {
+                        peer,
+                        addresses,
+                        old_peer,
+                        ..
+                    },
+                )) => {
+                    debug!(
+                        "Routing table updated. Peer: {peer:?}, Addresses: {addresses:?}, Old \
+                         peer: {old_peer:?}",
+                        peer = peer,
+                        addresses = addresses,
+                        old_peer = old_peer,
+                    );
+                    return Poll::Ready(Some(Event::RoutingUpdated(peer, addresses)));
+                }
+                #[cfg(feature = "dht")]
+                SwarmEvent::Behaviour(GossipnetBehaviourEvent::Kademlia(
+                    KademliaEvent::RoutablePeer {
+                        peer,
+                        address,
+                        ..
+                    },
+                )) => {
+                    debug!(
+                        "Routable peer: {peer:?}, Address: {address:?}",
+                        peer = peer,
+                        address = address,
+                    );
+                }
+                #[cfg(feature = "dht")]
```

Comment:
These interspersed conditional compilation flags make it tough to read. Please delegate handling of kadmelia and mdns events to separate methods